apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -36,6 +36,44 @@ namespace Google.Cloud.Spanner.V1
     /// </summary>
     public sealed partial class SessionPool
     {
+        /// <summary>
+        /// This struct contains information required to identify the database sessions.
+        /// It will be used as key in <see cref="_targetedPools"/>.
+        /// </summary>
+        internal readonly struct SessionPoolInfo
+        {
+            /// <summary>
+            /// Database name of the associated <see cref="TargetedSessionPool"/>.
+            /// </summary>
+            internal DatabaseName DatabaseName { get; }
+
+            /// <summary>
+            /// Database role of the associated <see cref="TargetedSessionPool"/>.
```

Comment:
I think we may find it simplest to guarantee that this isn't null - if we make it an empty string everywhere that there's no database role, it'll make all kinds of operations a bit simpler. I think it's reasonable to do that in the constructor, just with `DatabaseRole = databaseRole ?? ""`.